src_firmware/src/shell.c

```diff
@@ -105,6 +145,100 @@ static int cmd_off_on(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_position(const struct shell *shell, size_t argc, char *argv[])
+{
+	int ret;
+	uint32_t position;
+
+	if (argc == 1) {
+		ret = position_get(&position, relevant_channel);
+		if (ret == SUCCESS) {
+			shell_fprintf(shell, SHELL_NORMAL, "Position: %d\n", position);
+
+		} else if (ret == NOT_INITIALISED) {
+			shell_fprintf(shell, SHELL_ERROR,
+				"Driver not initialised, could't get position!\n");
+
+		} else {
+			shell_fprintf(shell, SHELL_ERROR, "Other error, code: %d\n", ret);
+		}
+
+		return 0;
+	} else if (argc == 2) {
+		position = (uint32_t)strtol(argv[1], NULL, 10);
+		ret = target_position_set(position, relevant_channel);
+		if (ret == SUCCESS) {
+			shell_fprintf(shell, SHELL_NORMAL, "Position set to: %d\n", position);
+
+		} else if (ret == NOT_INITIALISED) {
+			shell_fprintf(shell, SHELL_ERROR,
+				"Driver not initialised, could't set position!\n");
+
+		} else if (ret == DESIRED_VALUE_TO_HIGH) {
+			shell_fprintf(shell, SHELL_ERROR,
+				"Desired position to high! Desired: %d. Max: 360\n", position);
+
+		} else if (ret == UNSUPPORTED_FUNCTION_IN_CURRENT_MODE) {
+			shell_fprintf(shell, SHELL_ERROR,
+			"Function unsupported in current mode!\n");
+
+		} else {
+			shell_fprintf(shell, SHELL_ERROR, "Other error, code: %d\n", ret);
+		}
```

Comment:
There is lots of error code handling using if else statements. Consider creating a function to just map error code to string and pass it to be printed in shell. Strings can be generic as `could't set position!` is not needed - user knows which command he has just called ;)

Reply:
added TODO, but need to discuss how exactly this should be done